### PR TITLE
coerce_hash allow_nil and indifferent_access options

### PR DESCRIPTION
**Pull request by TSMMark**

### CODE REVIEW

File: spec/extensions/changes_spec.rb [hunk 1]
```diff
@@ -273,7 +273,7 @@
         end
 
         form = form_class.new(:foo => "12345")
-        assert_equal(Fixnum, new_value_class)
+        assert_equal(Integer, new_value_class)
```
Comment (TSMMark) [inline spec/extensions/changes_spec.rb:276]: https://weblog.rubyonrails.org/2016/10/8/this-week-in-rails-getting-ready-for-ruby-2-4/#:~:text=Fixnum%20and%20Bignum%20are%20deprecated,legacy%20classes%20were%20deprecated%20recently.

File: lib/gourami/coercer.rb [hunk 2]
```diff
@@ -120,21 +119,38 @@ def coerce_phone(value, options = {})
...
+      return if options[:allow_nil] && value.nil?
+
       hash_key_type = options[:key_type]
       hash_value_type = options[:value_type]
 
-      return {} unless value.is_a?(Hash) || (defined?(Sequel::Postgres::JSONHash) && value.is_a?(Sequel::Postgres::JSONHash))
+      hash_class = options[:indifferent_access] ? ActiveSupport::HashWithIndifferentAccess : Hash
+      hash = hash_class.new
+
+      return hash unless value.is_a?(Hash) || (defined?(Sequel::Postgres::JSONHash) && value.is_a?(Sequel::Postgres::JSONHash))
```
Comment (gzssince1990) [inline lib/gourami/coercer.rb:136]: On this line should we also check if `value.is_a?(ActiveSupport::HashWithIndifferentAccess)` or `ActiveSupport::HashWithIndifferentAccess` is basically a `Hash`?

Comment (TSMMark) [reply lib/gourami/coercer.rb:136]:
```
[3] Vydia(main)> ActiveSupport::HashWithIndifferentAccess.new.is_a?(Hash)
=> true
```

**Review by gzssince1990 — APPROVED** [review]
LGTM

**Review by tygee713 — APPROVED** [review]

**Pull request merged**